providers/UiProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import { createContext, useContext, useEffect, useState } from 'react';
+import { useNeynarProvider } from './NeynarProvider';
+import SignInDialog from '@/components/SignInDialog';
+import { useMediaQuery } from 'usehooks-ts';
+
+type UiContextType = {
+  menuOpen: boolean;
+  setMenuOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  checkLoggedIn: () => boolean;
+  isMobile: boolean;
+};
+const UiContext = createContext<UiContextType>({} as any);
+
+export default function UiProvider({ children }: any) {
```

Comment:
this is more of a `SignInProvider` than a `UiProvider` right?
- is this only handling `sign in`? if yes, change the name to be more accurate.
- if there is more than just `sign in` here, what else is there?